### PR TITLE
WIP: email plugin implementation

### PR DESCRIPTION
**Pull request by Hobbitsloth**
Someone has to finish the email plugin

### CODE REVIEW

File: plugins/email/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EMailLogic.java [hunk 1]
```diff
@@ -49,19 +52,30 @@
...
+ * @author Patrick Singer, Felix Burk
  */
 @Service
 public class EMailLogic {
 
 	private Session session;
 	private Folder inbox;
+	
+	@Reference
+	private Properties configLoader;
```
Comment (Legion2) [inline plugins/email/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EMailLogic.java]: this is not the configLoader this is the config

File: plugins/email/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EMailSpeech.java [hunk 2]
```diff
@@ -35,8 +33,9 @@
 /**
  * Class that defines the speech commands for the email functionality and calls the logic methods
  * 
- * @author Patrick Singer
+ * @author Patrick Singer, Felix Burk
  */
+
 @Service(EMailSpeech.class)
```
Comment (Legion2) [inline plugins/email/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EMailSpeech.java]: remove the `@Service` annotation

File: plugins/email/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EmailLogicTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import de.unistuttgart.iaas.amyassist.amy.test.TestFramework;
+
+/**
+ * TODO: Description
+ * @author Felix Burk
+ */
+@ExtendWith(FrameworkExtension.class)
+public class EmailLogicTest {
+	
+	@Mock
```
Comment (Legion2) [inline plugins/email/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EmailLogicTest.java]: remove the `@Mock` annotation

File: plugins/email/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EmailLogicTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	
+	/**
+	 * more to come
+	 */
+	@Test
+	public void testHasNewMessages() {
+		boolean b = false;
+		b = this.emailLogic.hasUnreadMessages();
+	
+		System.out.println("success? " + b);
```
Comment (Legion2) [inline plugins/email/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EmailLogicTest.java]: use assertThat

File: plugins/email/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EMailSpeech.java [hunk 5]
```diff
@@ -108,14 +102,8 @@ public String readRecentMails(String... params) {
...
-	// works
 	@Grammar("send example mail")
 	public String sendExampleMail(String... params) {
-		try {
-			return this.logic.sendMail("st142778@stud.uni-stuttgart.de", "Mail From Amy", "Hello!");
-		} catch (MessagingException e) {
-			this.logger.error("Sending mail failed!", e);
-			return "nope";
-		}
+		return this.logic.sendMail("st142778@stud.uni-stuttgart.de", "Mail From Amy", "Hello!");
```
Comment (Legion2) [inline plugins/email/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EMailSpeech.java]: remove your email address

File: plugins/email/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EMailSpeech.java [hunk 6]
```diff
@@ -88,17 +85,14 @@ public String numberOfNewMails(String... params) {
...
-		String[] messages;
-		try {
-			messages = this.logic.printPlainTextMessages(Integer.parseInt(params[1]));
-			return String.join("\n\n", messages);
-		} catch (MessagingException e) {
-			this.logger.error("Something is wrong with the inbox folder.", e);
-			return "nope";
+		String message;
+		message = this.logic.printPlainTextMessages(Integer.parseInt(params[1]));
+		if(message.equals("")) {
```
Comment (Legion2) [inline plugins/email/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EMailSpeech.java]: `isEmpty`

File: plugins/email/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EMailLogic.java [hunk 7]
```diff
@@ -165,27 +216,19 @@ public String sendMail(String recipient, String subject, String message) throws
...
-		startSession(username, password);
-		openInboxReadOnly();
+		String username = this.configLoader.getProperty(EMAIL_USR_KEY);
+		String password = this.configLoader.getProperty(EMAIL_PW_KEY);
+
+		if(username != null && password != null) {
+			startSession(username, password);
+			openInboxReadOnly();
+		}else {
+			this.logger.error("properties file not found");
```
Comment (Legion2) [inline plugins/email/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EMailLogic.java]: properties file is found but empty

File: Amy_Core_Main.launch [hunk 8]
```diff
@@ -12,5 +12,5 @@
 <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="de.unistuttgart.iaas.amyassist.amy.core.Main"/>
 <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="amy-core"/>
 <stringAttribute key="org.eclipse.jdt.launching.SOURCE_PATH_PROVIDER" value="org.eclipse.m2e.launchconfig.sourcepathProvider"/>
-<stringAttribute key="org.eclipse.jdt.launching.WORKING_DIRECTORY" value="${workspace_loc:amy}"/>
+<stringAttribute key="org.eclipse.jdt.launching.WORKING_DIRECTORY" value="${workspace_loc:Amy}"/>
```
Comment (Legion2) [inline Amy_Core_Main.launch]: is this change needed?

File: plugins/email/.gitignore [hunk 9]
```diff
@@ -0,0 +1 @@
+/target/
```
Comment (Legion2) [inline plugins/email/.gitignore]: remove this gitignore

File: plugins/email/pom.xml [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+	<parent>
+		<groupId>de.unistuttgart.iaas.amyassist</groupId>
+		<artifactId>amy-plugin</artifactId>
+		<version>0.3.0-Snapshot</version>
+		<relativePath>../pom.xml</relativePath>
+	</parent>
+	<dependencies>
+		<!-- https://mvnrepository.com/artifact/javax.mail/mail -->
+		<dependency>
+    		<groupId>javax.mail</groupId>
```
Comment (Legion2) [inline plugins/email/pom.xml]: apply formatter

File: Amy_Core_Main.launch (hunk 8 above)
Comment (astyz) [reply Amy_Core_Main.launch]: It didn't work for me so I changed it back and now it works again
